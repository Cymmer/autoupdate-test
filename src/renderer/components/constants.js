export * from './Search/constants';
export * from './SectionList/constants';
export * from './TaskList/constants';
export * from './ProblemList/constants';
