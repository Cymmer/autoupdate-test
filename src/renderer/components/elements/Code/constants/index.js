export { default as codeSizes } from './codeSizes';
export { default as programmingLanguages } from './programmingLanguages';
export { getSyntaxHighlighterLanguageFromExtension } from './programmingLanguages';
