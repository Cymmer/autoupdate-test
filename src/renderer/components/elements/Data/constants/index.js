export { default as dataSizes } from './dataSizes';
