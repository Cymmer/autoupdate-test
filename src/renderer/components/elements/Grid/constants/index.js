export { default as gridTypes } from './gridTypes';
