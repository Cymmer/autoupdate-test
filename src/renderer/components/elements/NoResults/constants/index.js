export { default as noResultsTypes } from './noResultsTypes';
