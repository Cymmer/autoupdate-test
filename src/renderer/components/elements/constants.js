export * from './ActionsDropdown/constants';
export * from './Button/constants';
export * from './Code/constants';
export * from './Data/constants';
export * from './Dropdown/constants';
export * from './Grid/constants';
export * from './Info/constants';
export * from './Input/constants';
export * from './Modal/constants';
export * from './NoResults/constants';
export * from './RadioGroup/constants';
export * from './Spinner/constants';
export * from './Tabs/constants';
export * from './Tags/Tag/constants';
export * from './Text/constants';
export * from './TextArea/constants';
export * from './Tooltip/constants';
export * from './XTermWrapper/constants';
