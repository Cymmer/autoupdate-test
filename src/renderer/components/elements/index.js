/* eslint-disable import/no-cycle */
export { default as ActionsDropdown } from './ActionsDropdown';
export { default as Anchor } from './Anchor';
export { default as Badge } from './Badge';
export { default as Breadcrumbs } from './Breadcrumbs';
export { default as Button } from './Button';
export { default as ButtonGroup } from './Button/Group';
export { default as IconButton } from './Button/IconButton';
export { default as IconLink } from './Button/IconLink';
export { default as ButtonLink } from './Button/Link';
export { default as Card } from './Card';
export { default as CardLink } from './Card/Link';
export { default as Checkbox } from './Checkbox';
export { default as CircularProgressBar } from './CircularProgressBar';
export { default as Code } from './Code';
export { default as CodeEditor } from './CodeEditor';
export { default as Cody } from './Cody';
export { default as Compiler } from './Compiler';
export { default as Container } from './Container';
export { default as Data } from './Data';
export { default as ControlledDropdown } from './Dropdown/Controlled';
export { default as Grid } from './Grid';
export { default as Header } from './Header';
export { default as Icon } from './Icon';
export { default as Info } from './Info';
export { default as InteractiveCompiler } from './InteractiveCompiler';
export { default as ControlledInput } from './Input/Controlled';
export { default as Label } from './Label';
export { default as Locked } from './Locked';
export { default as Modal } from './Modal';
export { default as ModalTitle } from './Modal/Title';
export { default as NavUser } from './NavUser';
export { default as NoResults } from './NoResults';
export { default as NotificationsDropdown } from './NotificationsDropdown';
export { default as Notification } from './NotificationsDropdown/Notification';
export { default as OptionalField } from './OptionalField';
export { default as Pagination } from './Pagination';
export { default as ProgressBar } from './ProgressBar';
export { default as RadioGroup } from './RadioGroup';
export { default as RequiredField } from './RequiredField';
export { default as ScreenLoader } from './ScreenLoader';
export { default as Section } from './Section';
export { default as Separator } from './Separator';
export { default as Shine } from './Shine';
export { default as Spinner } from './Spinner';
export { default as SubInfo } from './SubInfo';
export { default as Switch } from './Switch';
export { default as Tabs } from './Tabs';
export { default as Tags } from './Tags';
export { default as Text } from './Text';
export { default as Span } from './Text/Span';
export { default as ControlledTextArea } from './TextArea/Controlled';
export { default as Tooltip } from './Tooltip';
export { default as UserImage } from './UserImage';
export { default as XTermWrapper } from './XTermWrapper';
