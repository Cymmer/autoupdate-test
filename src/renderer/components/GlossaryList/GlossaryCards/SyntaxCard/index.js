import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles.module.scss';

import { Text, Card, Code } from '../../../elements';
import { textTypes, programmingLanguages } from '../../../elements/constants';

const SyntaxCard = ({ title, codes, language }) => (
  <Card className={styles.GlossaryCard}>
    <div className={styles.GlossaryCard_title}>
      <Text
        className={styles.GlossaryCard_title_text}
        type={textTypes.HEADING.XXS}
      >
        {title}
      </Text>
    </div>
    <div className={styles.GlossaryCard_content}>
      {codes.map((code, index) => (
        <Code
          key={index}
          className={styles.GlossaryCard_content_code}
          language={language}
        >
          {code}
        </Code>
      ))}
    </div>
  </Card>
);

SyntaxCard.defaultProps = {
  language: null,
};

SyntaxCard.propTypes = {
  title: PropTypes.string.isRequired,
  codes: PropTypes.arrayOf(PropTypes.string).isRequired,
  language: PropTypes.oneOf([
    programmingLanguages.PYTHON,
    programmingLanguages.C,
    programmingLanguages.CPP,
  ]),
};

export default SyntaxCard;
