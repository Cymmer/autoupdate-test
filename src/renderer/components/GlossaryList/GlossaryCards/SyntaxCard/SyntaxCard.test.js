import React from 'react';
import { shallow } from 'enzyme';
import SyntaxCard from './index';
import { programmingLanguages } from '../../../elements/constants';

describe('RENDERING - GlossaryList > GlossaryCards > SyntaxCard', () => {
  it('should render with default props', () => {
    const component = shallow(<SyntaxCard codes={[]} title="Dummy Title" />);
    expect(component).toMatchSnapshot();
  });

  it('should render all the codes depending on the programming language passed', () => {
    let component = shallow(
      <SyntaxCard
        codes={['Code1', 'Code2']}
        language={programmingLanguages.PYTHON}
        title="Dummy Title"
      />
    );
    expect(component).toMatchSnapshot();

    component = shallow(
      <SyntaxCard
        codes={['Code1', 'Code2']}
        language={programmingLanguages.C}
        title="Dummy Title"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
