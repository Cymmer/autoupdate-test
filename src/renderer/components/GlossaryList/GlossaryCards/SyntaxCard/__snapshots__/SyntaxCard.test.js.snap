// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - GlossaryList > GlossaryCards > SyntaxCard should render all the codes depending on the programming language passed 1`] = `
<Card
  className="GlossaryCard"
  id={null}
  onClick={null}
  onMouseEnter={null}
  onMouseLeave={null}
>
  <div
    className="GlossaryCard_title"
  >
    <Text
      className="GlossaryCard_title_text"
      colorClass="n___700"
      element={null}
      id={null}
      type="heading___xxs"
    >
      Dummy Title
    </Text>
  </div>
  <div
    className="GlossaryCard_content"
  >
    <Connect(Code)
      className="GlossaryCard_content_code"
      key="0"
      language="python"
    >
      Code1
    </Connect(Code)>
    <Connect(Code)
      className="GlossaryCard_content_code"
      key="1"
      language="python"
    >
      Code2
    </Connect(Code)>
  </div>
</Card>
`;

exports[`RENDERING - GlossaryList > GlossaryCards > SyntaxCard should render all the codes depending on the programming language passed 2`] = `
<Card
  className="GlossaryCard"
  id={null}
  onClick={null}
  onMouseEnter={null}
  onMouseLeave={null}
>
  <div
    className="GlossaryCard_title"
  >
    <Text
      className="GlossaryCard_title_text"
      colorClass="n___700"
      element={null}
      id={null}
      type="heading___xxs"
    >
      Dummy Title
    </Text>
  </div>
  <div
    className="GlossaryCard_content"
  >
    <Connect(Code)
      className="GlossaryCard_content_code"
      key="0"
      language="c"
    >
      Code1
    </Connect(Code)>
    <Connect(Code)
      className="GlossaryCard_content_code"
      key="1"
      language="c"
    >
      Code2
    </Connect(Code)>
  </div>
</Card>
`;

exports[`RENDERING - GlossaryList > GlossaryCards > SyntaxCard should render with default props 1`] = `
<Card
  className="GlossaryCard"
  id={null}
  onClick={null}
  onMouseEnter={null}
  onMouseLeave={null}
>
  <div
    className="GlossaryCard_title"
  >
    <Text
      className="GlossaryCard_title_text"
      colorClass="n___700"
      element={null}
      id={null}
      type="heading___xxs"
    >
      Dummy Title
    </Text>
  </div>
  <div
    className="GlossaryCard_content"
  />
</Card>
`;
